File: csharp/DiagramPrinter/DiagramPage.cs
namespace DiagramPrinter;

public class DiagramPage
{
    public string Name { get; }
    public string PageContents { get; }

    public DiagramPage(string name, string pageContents)
    {
        Name = name;
        PageContents = pageContents;
    }

    public void AddDataFrom(FlowchartReportItems data)
    {
        try
        {
            if (PageContents.Contains("{0}"))
                string.Format(PageContents, data.ToArray());
        }
        catch (System.FormatException e)
        {
        }
        
    }
}

File: java/src/main/java/sammancoaching/DiagramPage.java
package sammancoaching;


import java.util.IllegalFormatException;

public class DiagramPage {
    private final String name;
    private String pageContents;

    public DiagramPage(String name, String pageContents) {
        this.name = name;
        this.pageContents = pageContents;
    }

    public String getName() {
        return name;
    }

    public String getPageContents() {
        return pageContents;
    }

    public void addDataFrom(FlowchartReportItems data) {
        try {
            if (pageContents.contains("{0}")) {
                pageContents = String.format(pageContents, data.toArray());
            }
        } catch (IllegalFormatException e) {

        }
    }
}
